src/main/java/com/microsoft/azure/functions/binding/RpcComplexDataSources.java

```diff
@@ -112,7 +113,7 @@ private HttpRequestMessageImpl(RpcHttpRequestDataSource parentDataSource, Object
         @Override
         public String getMethod() { return this.parentDataSource.httpPayload.getMethod(); }
         @Override
-        public Map<String, String> getHeaders() { return this.parentDataSource.httpPayload.getHeadersMap(); }
+        public Map<String, TypedData> getHeaders() { return this.parentDataSource.httpPayload.getHeadersMap(); }
```

Comment:
Don't do this because you are exposing an internal data structure to the public developers. Try to get all the actual values (like String, int, etc.) and return a Map with those values.

And make sure the return type matches the [signature here](https://github.com/Azure/azure-functions-java-library/pull/10/files#diff-e6d1900cf8bd12c64341fdc11d63cc43R9).

Reply:
Isn't RpcComplexDataSources.java internal to worker? How is this exposed to user? I have updated the signature to  Map<String, Object> getHeaders() in azure-functions-java-library.

Reply:
Also aren't HttpRequestMessage and HttpResponseMessage are internal implementation details of worker? If yes, can we move them here? If they need to be exposed to user, can you please provide an example?

Reply:
@JunyiYi -  I think you are right. If the requirement for http headers is to be <string, string>, we should not be changing RpcHttp in proto file. I will look into this and update the PR soon.

Reply:
`HttpRequestMessage` is a public interface, it contains a method definition [`Map<String, Object> getHeaders()`](https://github.com/Azure/azure-functions-java-library/pull/10/files#diff-e6d1900cf8bd12c64341fdc11d63cc43R34). And you are returning `TypedData` here (which is not correct actually, because it needs to `@Override` the correct types). The scope in Java is just a compilation stage thing, when it is running, user could access the `TypedData` you are returning, and I think it should be an internal data structure (because it is a data structure used in GRPC).

So rather than returning the `TypedData`, a better way is to convert it to Java native types (or even POJO type), and return it to the user.